staging/src/k8s.io/apiserver/pkg/server/httplog/httplog.go

```diff
@@ -135,6 +135,15 @@ func StatusIsNot(statuses ...int) StacktracePred {
 	}
 }
 
+// StatusNotAmong returns a StacktracePred which will cause stacktraces to be logged
+// for any status *not* in the given map.
+func StatusNotAmong(statuses map[int]bool) StacktracePred {
```

Comment:
Instead of introducing a new function, why not update `StatusIsNot` to be more perfomant by taking the provided statuses and constructing a map to use when the `StacktracePred` is called?

Reply:
StatusIsNot may be passed various sequences of statuses.
The change in this PR optimizes the case where the statuses are known at compilation time where the map is constructed once.

If we take the give statuses on the fly and construct the map, I think it is not more performant - both are of O(N) complexity.

Thanks

Reply:
> StatusIsNot may be passed various sequences of statuses.
> The change in this PR optimizes the case where the statuses are known at compilation time where the map is constructed once.
> 
> If we take the give statuses on the fly and construct the map, I think it is not more performant - both are of O(N) complexity.

What I mean is, update `StatusIsNot` to something like:

```go
func StatusIsNot(statuses ...int) StacktracePred {
        // construct map
	return func(status int) bool {
		// map lookup
	}
}
```
Then you can create the StacktracePred just once and use it, e.g.:

```
var statusesNoTracePred = httplog.StatusIsNot(..._)
```

Reply:
I have modified the code along the lines above.